### DISCUSSION TITLE
How to know if the current buffer is an output buffer?

### DISCUSSION THREAD

**Discussion by exelotl** [orbitalquark/textadept#460] [Q&A] [answered]
Some time ago I mapped `ctrl+w` to close the current buffer. As a convenience, it will unsplit the view if it detects that you closed the output buffer (since the output buffer always appears in a new pane which is annoying to close.)

```lua
local ctl = function (k)
	return (not OSX and "ctrl+" or "cmd+")..k
end

-- closing the message buffer will hide the view
keys[ctl"w"] = function ()
	local label = buffer.tab_label
	local closed = buffer:close()
	if closed and label == "[Message Buffer]" then
		view:unsplit()
	end
end
```

Since updating to Textadept 12.x, this fails because `buffer.tab_label` is _write only_. 

Is there any other way to make this work?

**Comment by rgieseke** [comment, marked as answer]
You could check the `buffer._type`.

**Reply by rgieseke** [reply]
Not sure if it's documented, but it's tested here (and it's also used in Textredux):

https://github.com/orbitalquark/textadept/blob/89dfa1ec97ea9ad271cc2366ef5086ab620528e0/test/test.lua#L1024

**Reply by exelotl** [reply]
Ah brilliant! Yeah, I couldn't find it in the docs, but it works:

```lua
-- closing the message buffer will hide the view
keys[ctl"w"] = function ()
	local type = buffer._type
	local closed = buffer:close()
	if closed and type == "[Output Buffer]" then
		view:unsplit()
	end
end
````

Thanks a bunch!
